Lib/tempfile.py

```diff
@@ -121,6 +127,12 @@ def _sanitize_params(prefix, suffix, dir):
             prefix = template
         else:
             prefix = _os.fsencode(template)
+    if output_type is str:
+        if any(sep in prefix for sep in _path_separators):
```

Comment:
You can simply test that `os.path.dirname(prefix+suffix)` is empty.

Reply:
I would also prefer to reuse dirname() function.